compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -826,9 +855,19 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(implicit ctx: Context) {
         }
         if (!isImplicit) caseBindingsBuf += scrutineeBinding
         val gadtCtx = typer.gadtContext(gadtSyms).addMode(Mode.GADTflexible)
-        val pat1 = typer.typedPattern(cdef.pat, scrutType)(gadtCtx)
-        if (reducePattern(caseBindingsBuf, scrutineeSym.termRef, pat1)(gadtCtx) && guardOK)
-          Some((caseBindingsBuf.toList, cdef.body))
+        val fromBuf = mutable.ListBuffer.empty[TypeSymbol]
+        val toBuf = mutable.ListBuffer.empty[TypeSymbol]
+        if (reducePattern(caseBindingsBuf, fromBuf, toBuf, scrutineeSym.termRef, cdef.pat)(gadtCtx) && guardOK) {
+          val caseBindings = caseBindingsBuf.toList
+          val from = fromBuf.toList
```

Comment:
Here I would opt against defining from, to separately. Have the test read `(fromBuf.isEmpty)` and convert to lists in the following `subst` call.

Reply:
`to` is used another time on a line below, so it seems to me it's better if we just extract both results from the buffers at the same time.

Reply:
Ah I overlooked that